tests/End2EndTestCase.php

```diff
@@ -0,0 +1,79 @@
+<?php
+/**
+ * @copyright 2015-2017 Contentful GmbH
+ * @license   MIT
+ */
+
+namespace Contentful\Tests\Delivery;
+
+use Contentful\Delivery\Client;
+use Contentful\Log\ArrayLogger;
+
+class End2EndTestCase extends \PHPUnit_Framework_TestCase
+{
+    /**
+     * @var string
+     */
+    protected static $cacheDir = '';
+
+    public static function setUpBeforeClass()
+    {
+        self::$cacheDir = __DIR__ . '/../build/cache';
+    }
+
+    /**
+     * @param string $key
+     *
+     * @return Client
+     */
+    protected function getClient($key)
+    {
+        $options = getenv('CONTENTFUL_CDA_SDK_TESTING_URL')
+            ? ['uriOverride' => getenv('CONTENTFUL_CDA_SDK_TESTING_URL')]
+            : [];
+
+        switch ($key) {
```

Comment:
This looks much better. :) Any particular reason you went for a switch statement instead of many methods?

One advantage of methods would be that we could parameterize them which would cut down on the number of different methods/cases we need and we could inject things like the logger or cache path at that point.

Though overall not a big difference.